### PR TITLE
BUG: support pandas objects in iloc with old numpy versions

### LINKED ISSUE #17193: passing Index or Series to iloc fails with old numpy
#### Code Sample, a copy-pastable example if possible

```python
>>> s = pd.Series([1,2])    
>>> s.iloc[pd.Series([0])] = 2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/core/indexing.py", line 198, in __setitem__
    self._setitem_with_indexer(indexer, value)
  File "pandas/core/indexing.py", line 619, in _setitem_with_indexer
    value=value)
  File "pandas/core/internals.py", line 3313, in setitem
    return self.apply('setitem', **kwargs)
  File "pandas/core/internals.py", line 3201, in apply
    applied = getattr(b, f)(**kwargs)
  File "pandas/core/internals.py", line 864, in setitem
    values[indexer] = value
IndexError: unsupported iterator index
>>> np.version.version
'1.7.0'
```
#### Problem description

This is a consequence of #6168 , which is fixed in more recent numpy versions (but 1.7.0 is still a supported version).

Not 100% sure this is worth fixing (i.e. how interested we are in keeping compatibility with numpy 1.8.0), but the fix is trivial, so I'll just push a PR and let you judge.

#### Expected Output

Just no error.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.0-3-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.UTF-8
LOCALE: None.None

pandas: 0+unknown
pytest: 3.2.0
pip: None
setuptools: 36.2.7
Cython: None
numpy: 1.7.0
scipy: None
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 1.5
pytz: 2012c
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
>>> 

</details>

### PR DESCRIPTION
**Pull request by toobaz**
- [x] closes #17193
 - [x] tests added / passed
 - [x] passes ``git diff master -u -- "*.py" | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

**Comment by gfyoung** [comment]
@toobaz : I'm not sure if you saw my comment, but we intend to drop support for `numpy` < 1.9 in this upcoming release per #17002.

**Comment by toobaz** [comment]
> @toobaz : I'm not sure if you saw my comment, but we intend to drop support for numpy < 1.9 in this upcoming release per #17002.

Yes, sorry, I saw it now only. Shall I rebase just leaving the new test? It makes sense regardless of numpy versions.

**Comment by gfyoung** [comment]
@toobaz : No worries!  I'm a little hesitant to accept your "patch" (i.e. the change in the non-test file) if it's not necessary for newer versions of `numpy`.  However, I think we would be perfectly fine just accepting the test (though we wouldn't merge until after #17002).

**Comment by toobaz** [comment]
> However, I think we would be perfectly fine just accepting the test (though we wouldn't merge until after #17002).

Sure, that's what I meant: I will then wait for #17002 (or at least for the numpy version used in circleci to change) and then rebase leaving the test only.

**Comment by gfyoung** [comment]
Ah, okay.  We're on the same page here.  Absolutely.

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -278,6 +278,7 @@ Indexing
 - Fixes bug where indexing with ``np.inf`` caused an ``OverflowError`` to be raised (:issue:`16957`)
 - Bug in reindexing on an empty ``CategoricalIndex`` (:issue:`16770`)
 - Fixes ``DataFrame.loc`` for setting with alignment and tz-aware ``DatetimeIndex`` (:issue:`16889`)
+- Fixes ``.iloc`` with numpy 1.7 or 1.8 when passed a pandas object (:issue:`17193`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: this is not very descriptive. pls revise. you can just say older numpy.

File: pandas/core/internals.py [hunk 2]
```diff
@@ -861,6 +861,8 @@ def _is_empty_indexer(indexer):
 
         # set
         else:
+            # Work around GH 6168 to support numpy 1.7 and 1.8
+            indexer = getattr(indexer, 'values', indexer)
```
Comment (jreback) [inline pandas/core/internals.py]: ``indexer = np.asarray(indexer)`` might be better here

File: pandas/tests/indexing/test_iloc.py [hunk 3]
```diff
@@ -282,6 +282,19 @@ def test_iloc_setitem_list(self):
             index=["A", "B", "C"], columns=["A", "B", "C"])
         tm.assert_frame_equal(df, expected)
 
+    def test_iloc_setitem_pandas_object(self):
+        # GH 17193
```
Comment (jreback) [inline pandas/tests/indexing/test_iloc.py]: specify that this is for older numpy

File: pandas/core/internals.py (hunk 2 above)
Comment (jreback) [inline pandas/core/internals.py]: you can be explcit and use _np_version_under1p19 (then this is easier to remove later). alternatively the ``np.asarray`` may work.

**Review by jreback — CHANGES_REQUESTED** [review]
fine with this.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by toobaz** [comment]
> you can be explcit and use _np_version_under1p19 (then this is easier to remove later).

done

> alternatively the np.asarray may work.

Unfortunately fails when ``indexer`` is a slice

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @toobaz

**Pull request merged**